### PR TITLE
Added "Head First Java 2nd edition"

### PR DESCRIPTION
**Pull request by kclalit64**
PDF format of "Head First Java 2nd edition"

## What does this PR do?
Add resource(s)

### Description
Head First Java 2nd Edition
### Why is this valuable (or not)?
s a widely acclaimed book used by beginners to Java programming using lucid and eli5 approaches to concepts
### How do we know it's really free?
It's hosted by an organisation for free of cost
### For book lists, is it a book? For course lists, is it a course? etc.
It's a book
## Checklist:
- [x] Read our [contributing guidelines](https://github.com/EbookFoundation/free-programming-books/blob/main/docs/CONTRIBUTING.md).
- [x] [Search](https://ebookfoundation.github.io/free-programming-books-search/) for duplicates.
- [x] Include author(s) and platform where appropriate.
- [x] Put lists in alphabetical order, correct spacing.
- [x] Add needed indications (PDF, access notes, under construction).
- [x] Used an informative name for this pull request.

## Follow-up

- Check the status of GitHub Actions and resolve any reported warnings!

### CODE REVIEW

File: books/free-programming-books-langs.md [hunk 1]
```diff
@@ -1017,6 +1017,7 @@ That section got so big, we decided to split it into its own file, the [BY SUBJE
 * [Data Structures in Java for the Principled Programmer (2007)](https://web.archive.org/web/20190302130416/http://dept.cs.williams.edu/~bailey/JavaStructures/Book_files/JavaStructures.pdf) - Duane A. Bailey (PDF)
 * [Essential Java](https://www.programming-books.io/essential/java/) - Krzysztof Kowalczyk, StackOverflow Contributors
 * [Google's Java Style Guide](https://google.github.io/styleguide/javaguide.html)
+* [Head First Java (2nd Edition)](https://www.rcsdk12.org/cms/lib/NY01001156/Centricity/Domain/4951/Head_First_Java_Second_Edition.pdf) - Kathy Sierra, Bert Bates (PDF)
```
Comment (shrn01) [inline books/free-programming-books-langs.md:1020]: Is this an authorized distribution?

Comment (eshellman) [reply books/free-programming-books-langs.md:1020]: no
